src/router.ts

```diff
@@ -394,35 +411,66 @@ export async function handleNewVoter(
     userInfo.isDemo
   );
 
-  let slackChannelName = userInfo.isDemo ? 'demo-lobby' : 'lobby';
-  if (entryPoint === LoadBalancer.PULL_ENTRY_POINT) {
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): New voter will enter Slack channel: ${slackChannelName}`
-    );
-  } else if (entryPoint === LoadBalancer.PUSH_ENTRY_POINT) {
-    userInfo.stateName = LoadBalancer.getPushPhoneNumberState(
-      twilioPhoneNumber
-    );
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): Determined that twilioPhoneNumber ${twilioPhoneNumber} corresponds to U.S. state ${userInfo.stateName} based on hard coding in LoadBalancer.`
-    );
-    const selectedChannelName = await LoadBalancer.selectSlackChannel(
-      redisClient,
-      LoadBalancer.PUSH_ENTRY_POINT,
-      userInfo.stateName
-    );
+  let slackChannelName = null as string | null;
 
-    logger.debug(
-      `ROUTER.handleNewVoter (${userInfo.userId}): LoadBalancer returned Slack channel ${selectedChannelName} for new PUSH voter.`
-    );
-    if (selectedChannelName) {
-      slackChannelName = selectedChannelName;
-    } else {
-      // If LoadBalancer didn't find a Slack channel, then select #national or #demo-national as fallback.
-      slackChannelName = userInfo.isDemo ? 'demo-national' : 'national';
-      logger.error(
-        `ROUTER.handleNewVoter (${userInfo.userId}): ERROR LoadBalancer did not find a Slack channel for new PUSH voter. Using ${slackChannelName} as fallback.`
+  // do we already know the voter's state?
+  const knownState = await DbApiUtil.getKnownPhoneState(
+    userOptions.userPhoneNumber
+  );
+  if (knownState) {
+    const stateName = StateParser.determineState(knownState);
+    if (stateName) {
+      userInfo.stateName = stateName;
+      slackChannelName = await LoadBalancer.selectSlackChannel(
+        redisClient,
+        LoadBalancer.PULL_ENTRY_POINT,
+        stateName,
+        userInfo.isDemo
       );
+      if (slackChannelName) {
+        logger.info(
+          `ROUTER.handleNewVoter (${userInfo.userId}): New voter is in known state {stateName}, channel {slackChannelName}`
```

Comment:
instead of `channel {slackChannelName}` can we say something like `selected Slack channel is {slackChannelName}`